docs/dev/README.md

```diff
@@ -0,0 +1,110 @@
+# ModelarDB development
+We are happy to receive contributions in the form of pull requests. This document describes the structure of the code
+and general considerations to consider when doing further development. As such, this document should be used as a guideline
+when contributing to the repository. 
+
+## Structure
+The ModelarDB project consists of the following crates:
+
+* [modelardb_client](https://github.com/ModelarData/ModelarDB-RS/tree/master/crates/modelardb_client) - ModelarDB's command-line client in the form of the binary `modelardb`.
+* [modelardb_common](https://github.com/ModelarData/ModelarDB-RS/tree/master/crates/modelardb_common) - Library providing shared functions, macros, and types for use by the other crates.
+* [modelardb_compression](https://github.com/ModelarData/ModelarDB-RS/tree/master/crates/modelardb_compression) - Library providing lossless and lossy model-based compression of time series.
+* [modelardb_compression_python](https://github.com/ModelarData/ModelarDB-RS/tree/master/crates/modelardb_compression_python) - Python interface for the modelardb_compression crate.
+* [modelardb_server](https://github.com/ModelarData/ModelarDB-RS/tree/master/crates/modelardb_server) - The ModelarDB server in the form of the binary `modelardbd`.
+
+## Components
+Each major component in the ModelarDB project is described in detail to support further development of the components 
+and ease integration between components. The ModelarDB project consists of the following major components:
+
+* [Arrow Flight API]() - TODO: Write documentation for Arrow Flight API.
```

Comment:
Maybe only use `-` or `*` for itemizes without numbers throughout the documentation for consistency.